### PR TITLE
remove getArgs in favor of node:util.parseArgs

### PR DESCRIPTION
**Pull request by Scrub1492**
See: https://github.com/themoeway/yomitan/pull/459#discussion_r1436680737

### CODE REVIEW

File: dev/bin/build.js [hunk 1]
```diff
@@ -247,14 +266,15 @@ export async function main(argv) {
...
+            // eslint-disable-next-line dot-notation
+            argv.length === 0 || args['all'] ?
+            manifestUtil.getVariants().filter(({buildable}) => buildable !== false).map(({name}) => name) : []
         ));
         await build(buildDir, extDir, manifestUtil, variantNames, manifestPath, dryRun, dryRunBuildZip, yomitanVersion);
     } finally {
         // Restore manifest
-        const manifestName = /** @type {?string} */ ((!args.get('default') && args.get('manifest') !== null) ? args.get('manifest') : null);
+        // eslint-disable-next-line dot-notation
+        const manifestName = /** @type {?string} */ ((!args['default'] && typeof args['manifest'] !== 'undefined') ? args['manifest'] : null);
```
Comment (toasted-nutbread) [inline dev/bin/build.js:277]: Suggestions:
- Just use dot notation
- Switch the arg descriptors from using `kebab-case` to `camelCase` for better idiomatic JS
- Can the type cast be changed to a declaration annotation, which should be safer?
  ```js
  /** @type {?string} */
  const manifestName = ...;
  ```

File: dev/bin/build.js [hunk 2]
```diff
@@ -218,24 +219,42 @@ function ensureFilesExist(directory, files) {
...
+        },
+        'dry-run': {
+            type: 'boolean',
+            default: false
+        },
+        'dry-run-build-zip': {
+            type: 'boolean',
+            default: false
+        },
+        'yomitan-version': {
```
Comment (toasted-nutbread) [inline dev/bin/build.js:247]: `yomitan-version` - can this be made into just `version`? Will probably affect other places in the app/deployment process, but IMO the name of the extension should probably be omitted from a lot of the internal codebase.

Comment (toasted-nutbread) [reply dev/bin/build.js]: You can search codebase for regex `dry-run|yomitan-version` to find other files that need adjustment

File: dev/bin/build.js (hunk 1 above)
Comment (Scrub1492) [reply dev/bin/build.js]: Typescript is not too happy about the typecast
```
Type 'string | boolean | (string | boolean)[] | null' is not assignable to type 'string | null'.
  Type 'boolean' is not assignable to type 'string'.ts(2322)
```

Comment (toasted-nutbread) [reply dev/bin/build.js]: You can leave it as-is then, I wasn't sure how much typing util.parseArgs provided

**Comment by Scrub1492** [comment]
Switching to camel case means that you have to type in camel case in the command line, is this desirable?

**Comment by toasted-nutbread** [comment]
> Switching to camel case means that you have to type in camel case in the command line, is this desirable?

Yeah, there's no real reason to prefer one or the other for cmd line.

**Review by djahandarie — APPROVED** [review]

**Pull request merged**